packages/core/src/messages/responses/cache-batch-get.ts

```diff
@@ -0,0 +1,97 @@
+import {SdkError} from '../../errors';
+import {ResponseBase, ResponseError, ResponseSuccess} from './response-base';
+import {truncateString} from '../../internal/utils';
+import {CacheGet} from '../..';
+
+/**
+ * Parent response type for a cache get batch request.  The
+ * response object is resolved to a type-safe object of one of
+ * the following subtypes:
+ *
+ * - {Success}
+ * - {Error}
+ *
+ * `instanceof` type guards can be used to operate on the appropriate subtype.
+ * @example
+ * For example:
+ * ```
+ * if (response instanceof BatchGet.Error) {
+ *   // Handle error as appropriate.  The compiler will smart-cast `response` to type
+ *   // `BatchGet.Error` in this block, so you will have access to the properties
+ *   // of the Error class; e.g. `response.errorCode()`.
+ * }
+ * ```
+ */
+export abstract class Response extends ResponseBase {
+  public values(): (string | undefined)[] | undefined {
+    if (this instanceof Success) {
+      return (this as Success).values();
+    }
+    return undefined;
+  }
+
+  public results(): CacheGet.Response[] | undefined {
+    if (this instanceof Success) {
+      return (this as Success).results();
+    }
+    return undefined;
+  }
+}
+
+class _Success extends Response {
+  private readonly body: CacheGet.Response[];
+  constructor(body: CacheGet.Response[]) {
+    super();
+    this.body = body;
+  }
+
+  /**
+   * Returns the data as a list of utf-8 strings, decoded from the underlying byte array.
+   * May contain nulls corresponding to cache misses.
+   * @returns {(string | undefined)[]}
+   */
+  public values(): (string | undefined)[] {
```

Comment:
I wonder if we should consider making the return type for this be a `Record<string, string | uint>`, or a `Map`.

I can definitely imagine cases where a user would just want them in order, but they could maybe achieve that sufficiently through `results`. It feels like the more natural / common / simple use case (which is what we want to prioritize for this `.values` function) might be to get back a structure that they can index into by key.

would really love to get other folks thoughts on this though, especially since is the first major SDK we're releasing this in and we will be treating it as a canonical example.

Reply:
yeah I like that idea considering they provided a Record or Map, they're probably expecting a similar structure back.  

I had missed this before but it seems like [dictionaryGetFields](https://github.com/momentohq/client-sdk-javascript/blob/main/packages/core/src/messages/responses/cache-dictionary-get-fields.ts) returns items as Record or Map as well

Reply:
💯 on the record/map as the default. In the past when we have exposed an array (eg in earlier iterations of `dictionaryGetFields`), I believe we found the record/map easier to use. My intuition is users reach for the record/map first as well.